### PR TITLE
Bugfix: remove umountDuplicate and add warning event

### LINKED ISSUE #3399: [BUG] CSI Fuse Recovery fails after app container restarts
**What is your environment(Kubernetes version, Fluid version, etc.)**
Both Fluid v0.9 and v1.0 branch suffers this bug. 

**Describe the bug**
After an app container restarts, any Fuse daemon restarts cannot be recovered by Fluid's CSI Plugin. In app container, it reports error like `Transport Endpoint is not connected`.

**What you expect to happen:**
Fuse recovery success.

**How to reproduce it**
1. Create Fluid Dataset and Runtime
2. Create app pod with Fluid Dataset mounting on it (must use `HostToContainer` mountPropagation to make Fuse recovery work )
3. Delete or kill FUSE pod to trigger FUSE recovery.
4. FUSE recovery succeeds
5. Kill app container in the app pod to make it restarts
6. Delete or kill FUSE pod to trigger FUSE recovery.
7. According to Dataset's events, FUSE recovery successfully works but app container fails to access data with `Transport Endpoint not connected` error.

**Additional Information**

### PR DESCRIPTION
**Pull request by wangshli**
### Ⅰ. Describe what this PR does
Remove umountDuplicate when recovering and add warning event if mountpoint has been mounted too many times. 

### Ⅱ. Does this pull request fix one issue?

fixes #3399

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: pkg/csi/recover/recover.go [hunk 1]
```diff
@@ -118,15 +121,24 @@ func NewFuseRecover(kubeClient client.Client, recorder record.EventRecorder, api
...
 		}
 	}
+	var recoverWarningThreshold int
+	if os.Getenv(RecoverWarningThreshold) != "" {
+		if recoverWarningThreshold, err = strconv.Atoi(os.Getenv(RecoverWarningThreshold)); err != nil {
+			return nil, errors.Wrap(err, "got error when parsing recover warning threshold")
+		}
+	} else {
+		recoverWarningThreshold = defaultRecoverWarningThreshold
+	}
```
Comment (TrafalgarZZZ) [inline pkg/csi/recover/recover.go:131]: Please replace the logic with `utils.GetIntValueFromEnv`

File: pkg/csi/recover/recover.go [hunk 2]
```diff
@@ -41,10 +41,12 @@ import (
 )
 
 const (
-	defaultKubeletTimeout     = 10
-	defaultFuseRecoveryPeriod = 5 * time.Second
-	serviceAccountTokenFile   = "/var/run/secrets/kubernetes.io/serviceaccount/token"
-	FuseRecoveryPeriod        = "RECOVER_FUSE_PERIOD"
+	defaultKubeletTimeout          = 10
+	defaultFuseRecoveryPeriod      = 5 * time.Second
+	defaultRecoverWarningThreshold = 10
```
Comment (TrafalgarZZZ) [inline pkg/csi/recover/recover.go:46]: 10 may be too small for default recover warning threshold. How about 50?

File: pkg/csi/recover/recover.go [hunk 3]
```diff
@@ -158,7 +170,8 @@ func (r FuseRecover) recover() {
 
 	for _, point := range brokenMounts {
 		glog.V(4).Infof("Get broken mount point: %v", point)
-		r.umountDuplicate(point)
+		// do not umountDuplicate because if app container restart, umount duplicate mount may lead to recover successed but can not access data
```
Comment (TrafalgarZZZ) [inline pkg/csi/recover/recover.go:173]: You can add comments with the issue link.

**Review by TrafalgarZZZ — CHANGES_REQUESTED** [review]

File: pkg/csi/recover/recover.go (hunk 3 above)
Comment (wangshli) [reply pkg/csi/recover/recover.go:173]: ok.

File: pkg/csi/recover/recover.go (hunk 1 above)
Comment (wangshli) [reply pkg/csi/recover/recover.go]: done.

File: pkg/csi/recover/recover.go (hunk 2 above)
Comment (wangshli) [reply pkg/csi/recover/recover.go]: done.

File: pkg/csi/recover/recover.go [hunk 4]
```diff
@@ -213,5 +217,14 @@ func (r *FuseRecover) eventRecord(point mountinfo.MountPoint, eventType, eventRe
 		return
 	}
 	glog.V(4).Infof("record to dataset: %s, namespace: %s", dataset.Name, dataset.Namespace)
-	r.Recorder.Eventf(dataset, eventType, eventReason, "Fuse recover %s succeed", point.MountPath)
+	if eventReason == common.FuseRecoverSucceed {
+		r.Recorder.Eventf(dataset, eventType, eventReason, "Fuse recover %s succeed", point.MountPath)
+	}
+	if eventReason == common.FuseRecoverFailed {
+		r.Recorder.Eventf(dataset, eventType, eventReason, "Fuse recover %s failed", point.MountPath)
+	}
```
Comment (TrafalgarZZZ) [inline pkg/csi/recover/recover.go:225]: How about `switch-case`?

Comment (wangshli) [reply pkg/csi/recover/recover.go]: done.

File: pkg/csi/recover/recover.go [hunk 5]
```diff
@@ -213,5 +217,14 @@ func (r *FuseRecover) eventRecord(point mountinfo.MountPoint, eventType, eventRe
...
-	r.Recorder.Eventf(dataset, eventType, eventReason, "Fuse recover %s succeed", point.MountPath)
+	switch eventReason {
+	case common.FuseRecoverSucceed:
+		r.Recorder.Eventf(dataset, eventType, eventReason, "Fuse recover %s succeed", point.MountPath)
+	case common.FuseRecoverFailed:
+		r.Recorder.Eventf(dataset, eventType, eventReason, "Fuse recover %s failed", point.MountPath)
+	}
+	// add warning event if point.count is larger than the threshold
+	if point.Count > r.recoverWarningThreshold {
+		r.Recorder.Eventf(dataset, corev1.EventTypeWarning, "TooManyRecovery", "Mountpoint %s has been mounted %v times", point.MountPath, point.Count)
```
Comment (TrafalgarZZZ) [inline pkg/csi/recover/recover.go:228]: Events may be automatically cleaned up by K8s, so better also log a message here to note that the point.Count is over than threshold.

**Pull request closed without merging**